lib/client.js

```diff
@@ -352,19 +351,10 @@ export default class Client {
       this.ready = true
       this._metadata = data.metadata
       this._context = data.context
-      this._iframe_session_timeout = data.iframe_session_timeout
-      this._idleState = new IdleState(data.iframe_session_timeout)
-      // Add an observer.  This will be called when the idleState changes
-      this._idleState.addObserver(state => {
-        if (state === 'active') {
-          // ... application is now active, timeout clock is reset to zero
-          // Send a postMessage indicating liveness + possibly a payload too
-          this.postMessage('session.live')
-        } else if (state === 'idle') {
-          // ... application is now idle, timeout clock is ticking
-          // Send a postMessage indicating sessionTimeout + possibly a payload too
-          this.postMessage('session.idle')
-        }
+      this._idleState = new IdleState()
+      this._idleState.addObserver(() => {
+        this.postMessage('session.live')
+        console.log('** post message sent: session.live')
       })
```

Comment:
Could I interest you in adding the debounce here instead? It would mean the IdleState class doesn't have to care about debouncing, and it can just call markActive whenever an event fires. ie. it makes debouncing a `client` concern, couples debouncing and post message logic together, and makes the intent a little clearer.
```suggestion
      this._idleState.addObserver(debounce(() => {
        this.postMessage('session.live')
        console.log('** post message sent: session.live')
      }, 1000))
```

Reply:
✅